fullstack-cert/python-projects/medical-data-validator-workshop/main.py

```diff
@@ -0,0 +1,82 @@
+import re
+
+medical_records = [
+    {
+        "patient_id": "P1001",
+        "age": 34,
+        "gender": "Female",
+        "diagnosis": "Hypertension",
+        "medications": ["Lisinopril"],
+        "last_visit_id": "V2301",
+    },
+    {
+        "patient_id": "P1002",
+        "age": 47,
+        "gender": "Male",
+        "diagnosis": "Type 2 Diabetes",
+        "medications": ["Metformin", "Insulin"],
+        "last_visit_id": "V2302",
+    },
+    {
+        "patient_id": "P1003",
+        "age": 29,
+        "gender": "Female",
+        "diagnosis": "Asthma",
+        "medications": ["Albuterol"],
+        "last_visit_id": "V2303",
+    },
+    {
+        "patient_id": "P1004",
+        "age": 56,
+        "gender": "Male",
+        "diagnosis": "Chronic Back Pain",
+        "medications": ["Ibuprofen", "Physical Therapy"],
+        "last_visit_id": "V2304",
+    },
+]
+
+
+def check_values(patient_id, age, gender, diagnosis, medications, last_visit_id):
```

Comment:
Name of the function is not very clear, this is particularly visible in what is returned from it. Without knowing it before, I'm not sure if I'd expect list of keys with wrong values from `check_values` function. Following the usage in `validate`, one idea for better name is `find_invalid_records`.